src/net_processing.cpp

```diff
@@ -2988,13 +2987,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
                   peer->m_starting_height, addrMe.ToString(), fRelay, pfrom.GetId(),
                   remoteAddr);
 
-        int64_t nTimeOffset = nTime - GetTime();
-        pfrom.nTimeOffset = nTimeOffset;
-        if (!pfrom.IsInboundConn()) {
-            // Don't use timedata samples from inbound peers to make it
-            // harder for others to tamper with our adjusted time.
-            AddTimeData(pfrom.addr, nTimeOffset);
-        }
+        pfrom.nTimeOffset = nTime - GetTime();
```

Comment:
Perhaps:

```c++
if (!pfrom.IsInboundConn() && std::abs(pfrom.nTimeOffset) >= 1800) {
    ++m_bad_non_inbound_time_offset;
    if (m_bad_non_inbound_time_offset == 3) {
        SetMiscWarning("Please check your computer's date and time are correct");
    }
}
``` 

(ie, add a counter to `PeerManagerImpl`, and on the third outbound connection that has a time offset of 30mins or more, set a warning that will show up in `getblockchaininfo`, `getnetworkinfo`, `getmininginfo` until restart)

Reply:
Alternatively, we regularly do new feeler and blocks-only connections, perhaps we could keep a rolling average of `abs(peer.nTimeOffset)` and report that in `getnetworkinfo`?